PermanentOrg/stela
# Add authentication middleware

In order to serve endpoints more substantial than a health check, stela's web server will need to be able to verify the authentication of callers. To this end, this commit adds middleware for verifying user JWTs and admin JWTs (in both cases using FusionAuth).

*Testing Instructions*
* Before testing, you'll have to add FusionAuth environment variables to your `.env` (see updates to README)
* Since there are no endpoints currently existing in this project that need authentication (just a health check), nothing is using the authentication middleware as of this commit. Thus, to test the middleware you'll have to add either
```
apiRoutes.use(verifyUserAuthentication)
```
or
```
apiRoutes.use(verifyAdminAuthentication)
```
to `src/routes/index.ts` right before the `apiRoutes.get` call.
* To test user authentication, add the appropriate middleware to the health check as described above. Observe that after that hitting the endpoint with no "Authorization" header results in a 401, but with an "Authorization" header of the form "Bearer: <A user JWT, such as that returned by `back-end`'s login endpoint>" it will work as usual
* To test admin authentication:
    1. Create a user in FusionAuth on the Local tenant
    2. Add a registration to the user for the admin-local application
    3. Hit FusionAuth's `/api/login` endpoint to get a JWT for that user
    4. Use that JWT to test `stela` admin authentication using the same process used for testing user authentication above
